test/test_utils/ec2.py

```diff
@@ -250,6 +250,7 @@ def check_instance_state(instance_id, state="running", region=DEFAULT_REGION):
     """
     instance_state = get_instance_state(instance_id, region)
     if state != instance_state:
+        terminate_instance(instance_id, region)
```

Comment:
Will this terminate the instance on first run of `get_instance_state`. Would this respect the @retry attempts?

Reply:
Yes, when the if condition is true, at line 253, we are raising exception if instance is not in running state and code go further anyway. So before before exception, I'm terminating ec2 instance when if condition is true.